core/storage/question/gae_models.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for storing the question data models."""
+
+from core.platform import models
+
+from google.appengine.ext import ndb
+
+import utils
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+class QuestionSnapshotMetadataModel(base_models.BaseSnapshotMetadataModel):
+    """Storage model for the metadata for a question snapshot."""
+    pass
+
+
+class QuestionSnapshotContentModel(base_models.BaseSnapshotContentModel):
+    """Storage model for the content of a question snapshot."""
+    pass
+
+
+class QuestionModel(base_models.VersionedModel):
+    """Model for storing Questions.
+
+    The id of instances of this class has the form
+    {{collection_id}}.{{random_hash_of_16_chars}}
+    """
+    SNAPSHOT_METADATA_CLASS = QuestionSnapshotMetadataModel
+    SNAPSHOT_CONTENT_CLASS = QuestionSnapshotContentModel
+    ALLOW_REVERT = True
+
+    # The title of the question.
+    title = ndb.StringProperty(required=True, indexed=True)
+    # The data for the question data.
+    question_data = ndb.JsonProperty(default={}, indexed=False)
```

Comment:
This should probably not have a default value, I think... There should be careful validation by the client to ensure that what's stored here is in the correct format.

Reply:
Updated.